packages/@aws-cdk/aws-synthetics/lib/test-script.ts

```diff
@@ -0,0 +1,115 @@
+import * as s3 from '@aws-cdk/aws-s3';
+import { Code } from './code';
+
+/**
+ * Specify a test that the canary should run
+ */
+export class Test {
+  /**
+   * Specify a custom test with your own code
+   *
+   * @returns `Test` associated with the specified Code object
+   * @param options The code and handler you want to run
+   */
+  public static custom(options: CustomOptions): Test {
+    Test.validateHandler(options.handler);
+    return new Test(undefined, options);
+  }
+
+  /**
+   * Use this method to access the template for heartBeat monitoring. This test will hit the specified url and report back
+   * information on the health of the website.
+   *
+   * @returns `Test` with the heartBeat template as the Code
+   * @param url The url of the website you want to test
+   *
+   * 🚧 TODO: implement
+   */
+  // public static heartBeat(url: string){}
+
+  /**
+   * Use this method to access the template for api Endpoint monitoring. This test will hit the endpoint specified and report
+   * back what it receives.
+   *
+   * @param url The url of the api endpoint you want to test
+   * @param options Options to customize the api endpoint template
+   *
+   * 🚧 TODO: implement
+   */
+  // public static apiEndpoint(url: string, options: apiOptions){}
+
+  /**
+   * Use this method to access the template for a broken link checker. This test will check a specified number of links and report
+   * back the first broken link found.
+   *
+   * @param url The starting url to check
+   * @param options Options to customzie the template
+   *
+   * 🚧 TODO: implement
+   */
+  // public static brokenLink(url: string, options: linkOptions){}
+
+  /**
+   * Verifies that the given handler ends in '.handler'. Returns the handler if successful and
+   * throws an error if not.
+   *
+   * @param handler - the handler given by the user
+   */
+  private static validateHandler(handler: string) {
+    if (!handler.endsWith('.handler')) {
```

Comment:
Must it really end with `.handler`? In Lambda for example, the handler is `<module>.<function>` and the user is free to name the function whatever they please.

Reply:
Yep -- it's gotta be `.handler`. [source](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-synthetics-canary-code.html#aws-properties-synthetics-canary-code-properties).

Reply:
WTF?!? So basically they are forcing you to pass a value... which you don't have a choice on. Nice.

Reply:
In any case... it **might** be interesting to hide the dirt from your API (just ask for a canary file name, if you can't infer or make one up), and append the `.handler` under the hood.

Reply:
I was thinking about this - the problem I foresee is that the user still needs to specify _which_ canary they want to run. i.e. the provided path contains both `canary1.js` and `canary2.js`. They then still need to specify whether the handler should be `canary1.handler` or `canary2.handler`.

All I'm doing here is saying "given a handler, check to see if the relevant file exists"